From fa32a0913b070c3d02fe502372ca34316a201888 Mon Sep 17 00:00:00 2001
From: Aman Achpal <aman.achpal@couchbase.com>
Date: Fri, 29 Sep 2017 16:31:39 +0530
Subject: [PATCH] MB-26187 Setup LD_LIBRARY_PATH environment variable

After updgrading v8 to 5.9 there are additional libraries
required by mapreduce_nif.so at runtime. Following the
conventions established in install/bin/couchbase-server,
we setup the LD_LIBRARY_PATH variable on Linux machines,
allowing the server to spawn successfully.

Change-Id: If063a7484a4c36e0f50badbd1711637da4be590f
---
 cluster_run | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/cluster_run b/cluster_run
index 29ff950d3..3fa627dd9 100755
--- a/cluster_run
+++ b/cluster_run
@@ -84,6 +84,16 @@ def os_specific(args, params):
             resource.setrlimit(resource.RLIMIT_NOFILE, (2048, 2048))
         params['env'] = {"ERL_MAX_PORTS": "2048"}
         params['env'].update(os.environ)
+    if platform.system() == 'Linux':
+        ## Configure LD_LIBRARY_PATH
+        LIB_PATH = PREFIX + '/lib'
+        MEMCACHED = LIB_PATH + '/memcached'
+        LD_LIBRARY_PATH = LIB_PATH + os.pathsep + MEMCACHED
+        env = os.environ.copy()
+        if 'LD_LIBRARY_PATH' in env:
+            LD_LIBRARY_PATH += (os.pathsep + env['LD_LIBRARY_PATH'])
+        env["LD_LIBRARY_PATH"] = LD_LIBRARY_PATH
+        params['env'] = env
 
 ebin_path = None
 cluster_extra_args = None
-- 
2.13.6 (Apple Git-96)

